server/src/main/java/org/elasticsearch/search/aggregations/AdaptingAggregator.java

```diff
@@ -40,14 +40,14 @@ public AdaptingAggregator(
          * agg tree. Thisis how it has always been and some aggs rely on it.
          */
         this.delegate = delegate.apply(subAggregators.fixParent(this));
-        assert this.delegate.parent() == parent : "invalid parent set on delegate";
+        assert this.delegate == null || this.delegate.parent() == parent : "invalid parent set on delegate";
```

Comment:
I originally added it as an assertion because I figured it'd only fail because of a programming error. But it can be a hard check.

And you are right about the null dance being weird. I'll have another look soon and see if I can avoid it. I like the simplicity of `buildOrNull` kinds of methods but it comes at a funny cost, that is for sure.